### ISSUE TITLE
UnusedImportsCheck does not handle qualified type names

### ISSUE THREAD

**Issue by cushon** [checkstyle/checkstyle#5044]
The following example contains a reference to `java.util.Map` in javadoc as the qualifier in `Map.Entry`, but checkstyle does not detect it as a use of the import.

```java
package test;

import java.util.Map;

class Test {
  /**
   * @see Map.Entry
   */
  public void f() {}
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="UnusedImports">
      <property name="processJavadoc" value="true"/>
    </module>
  </module>
</module>
```

```bash
$ java -jar checkstyle-8.2-all.jar -c checks.xml Test.java
...
[ERROR] Test.java:3:8: Unused import - java.util.Map. [UnusedImports]
```

**Comment by rnveach** [comment]
This is a duplicate of Issue #3453 .

**Comment by romani** [comment]
@cushon , this Check is unfortunate implementation based on regexp parsing of javadoc
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java
attention to `.... extends AbstractCheck`

such implementations are marked as deprecated till we/smb convert this implementation to use AST traversing that is more reliable.
Example of AST based Check: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
attention to `..... extends AbstractJavadocCheck {`

Chance that you/smb will fix this issue in the same design withtout extra regression is not that big.
I could allow this issue, but if on moment of regression we found unwanted/extra violations , I could withdraw approval and reject a fix. 
All are welcome to rewrite this Check from scratch.
